hydradx/model/amm/arbitrage_agent.py

```diff
@@ -438,3 +440,226 @@ def calculate_profit(init_agent, agent, asset_map={}):
     if sum([profit_asset[k] for k in profit_asset]) != sum([profit[k] for k in profit]):
         raise
     return profit
+
+
+def combine_swaps(
+        dex: OmnipoolState,
+        cex: dict[str, CentralizedMarket],
+        agent: Agent,
+        all_swaps: list[dict],
+        asset_map: dict[str, str],
+        max_liquidity: dict[str, dict[str, float]] = None
+):
+    # take the list of swaps and try to get the same result more efficiently
+    # in particular, make sure to buy *at least* as much of each asset as the net from the original list
+    exchanges = {**{ex_name: ex for (ex_name, ex) in cex.items()}, 'dex': dex}
+    ref_exchange: CentralizedMarket = exchanges['binance']
+    net_swaps = {}
+    return_swaps = []
+
+    if len(all_swaps) == 0:
+        return all_swaps
+    if 'dex' in all_swaps[0]:
+        all_swaps = flatten_swaps(all_swaps)
+
+    for ex_name, ex in exchanges.items():
+
+        test_agent = agent.copy()
+        test_ex = ex.copy()
+        default_swaps = list(filter(lambda s: s['exchange'] == ex_name, all_swaps))
+
+        # vvv With this uncommented, it will still recover most of the available gains
+        #
+        # if ex_name != 'dex':
```

Comment:
(not suggesting a change in this PR)

It would be good to implement a more elegant version of this where we can specify the exchanges to apply this logic to? Specifically it works better in Omnipool than for order books.